src/tusb_option.h

```diff
@@ -51,9 +51,29 @@
 #define OPT_MCU_SAMD21        200 ///< MicroChip SAMD21
 #define OPT_MCU_SAMD51        201 ///< MicroChip SAMD51
 
+// ST Synopsis OTG devices
 #define OPT_MCU_STM32F4       300 ///< ST STM32F4
-#define OPT_MCU_STM32F3       301 ///< ST STM32F3
 #define OPT_MCU_STM32H7       302 ///< ST STM32H7
+// ST FSDEV Devices
+#define OPT_MCU_STM32F0x0     330 ///< ST STM32F0x0

```

Comment:
why would you need this many of OPT_MCU. I would only prefer a family line such as OPT_MCU_STM32F0 / F1/F3 . They are 90% like each other within the line, these OPT_MCU_* is meant for USBD and higher layer to know which family high layer working with. For specific/detail mcu, the dcd_*.c could check the mcu specific macro for usage.

Reply:
> > @pigrew OK, I only read your comment [#123 (comment)](https://github.com/hathach/tinyusb/pull/123#discussion_r322715935) in this issue. Now I understand why you separate them. For B,C,D usb core, they are all sysnopsis, I think we can merge them together. Only the A stm32_fsdev is different. F1 is probably the only line has 102,103 for A and 105/107 for B. Maybe we can have 2 OPT_MCU for that `OPT_MCU_STM32F1x2_1x3` and `OPT_MCU_STM32F1x5_1x7`. Though since application should pick the correct dcd__.c, maybe it we can get away if #ifdef in the dcd__.c file
> 
> My preference would be to leave all STM32 as a 3-digit code for three reasons:
> 
> * Consistancy. If they are all [0-9]x[0-9], then I don't need to think about which define to use on STM32 since they all are using the same format.
> * Future-proof: It's possible that ST will extend a line (like the newer L or G) to have both the Synopsys and ST IP. I expect defines based on the first and third digits to work for the future.
> * I can include all of the _dcd drivers in my project at the moment. For my IDE, I include the entire "tusb/src" directory without needing to exclude certain drivers. If we need the user to only include the particular driver, then makefiles become more complicated.

Yeah, that make sense, though I still think it is too many, Though I think we should use the ST defined macros in both dcd_*.c to include the correct driver F102,103 vs F105,107.   

```
#if !defined (STM32F301x8) && !defined (STM32F302x8) && !defined (STM32F318xx) && \
    !defined (STM32F302xC) && !defined (STM32F303xC) && !defined (STM32F358xx) && \
    !defined (STM32F303x8) && !defined (STM32F334x8) && !defined (STM32F328xx) && \
    !defined (STM32F302xE) && !defined (STM32F303xE) && !defined (STM32F398xx) && \
    !defined (STM32F373xC) && !defined (STM32F378xx)
```